x-pack/plugins/notifications/server/plugin.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  PluginInitializerContext,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  Logger,
+} from '@kbn/core/server';
+import type {
+  NotificationsPluginSetupDeps,
+  NotificationsPluginStartDeps,
+  NotificationsPluginSetup,
+  NotificationsPluginStart,
+} from './types';
+import { EmailService } from './services';
+import { NotificationsConfigType } from './config';
+import { getResolvablePromise, type ResolvablePromise } from '../common/lib';
+import { PLUGIN_ID } from '../common';
+
+export class NotificationsPlugin
+  implements Plugin<NotificationsPluginSetup, NotificationsPluginStart>
+{
+  private readonly logger: Logger;
+  private readonly initialConfig: NotificationsConfigType;
+  private email: ResolvablePromise<EmailService>;
+  private emailConnector: string;
+
+  constructor(initializerContext: PluginInitializerContext<NotificationsConfigType>) {
+    this.logger = initializerContext.logger.get();
+    this.email = getResolvablePromise();
+    this.initialConfig = initializerContext.config.get();
+  }
+
+  public setup(core: CoreSetup, plugins: NotificationsPluginSetupDeps) {
+    // TODO this must be defaulted to 'elastic-cloud-email' for cloud
```

Comment:
Will we configure this in cloud deployment templates or hardcode in Kibana?

Reply:
In the cloud deployment template I'm guessing (hoping)?

Reply:
Yes, that's the idea, gonna remove the TODO line.

Reply:
This is addressed by https://github.com/elastic/cloud/pull/109221